specification/authorization/resource-manager/Microsoft.Authorization/preview/2024-02-01-preview/common-types.json

```diff
@@ -0,0 +1,520 @@
+          "items": {
+            "type": "string"
+          },
+          "description": "The type of operation."
+        },
+        "level": {
+          "type": "string",
+          "description": "The assignment level to which rule is applied."
+        },
+        "targetObjects": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of target objects."
+        },
+        "inheritableSettings": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of inheritable settings."
+        },
+        "enforcedSettings": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of enforced settings."
+        }
+      }
+    },
+    "RoleManagementPolicyRuleType": {
+      "type": "string",
+      "description": "The type of rule",
+      "enum": [
+        "RoleManagementPolicyApprovalRule",
+        "RoleManagementPolicyAuthenticationContextRule",
+        "RoleManagementPolicyEnablementRule",
+        "RoleManagementPolicyExpirationRule",
+        "RoleManagementPolicyNotificationRule",
+        "RoleManagementPolicyPimOnlyModeRule"
+      ],
+      "x-ms-enum": {
+        "name": "RoleManagementPolicyRuleType",
+        "modelAsString": true
+      }
+    },
+    "RoleManagementPolicyPimOnlyModeRule": {
+      "description": "The role management policy PIM only mode rule.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/RoleManagementPolicyRule"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "pimOnlyModeSettings": {
+          "$ref": "#/definitions/PIMOnlyModeSettings",
+          "description": "The PIM Only Mode settings"
+        }
+      }
+    },
+    "PIMOnlyModeSettings": {
+      "description": "The PIM Only Mode settings.",
+      "type": "object",
+      "properties": {
+        "mode": {
+          "type": "string",
+          "description": "Determines whether the setting is enabled, disabled or report only.",
+          "enum": [
+            "Disabled",
+            "Enabled",
+            "ReportOnly"
+          ],
+          "x-ms-enum": {
+            "name": "PIMOnlyMode",
+            "modelAsString": true
+          }
+        },
+        "excludes": {
+          "type": "array",
+          "description": "The list of excluded entities that the rule does not apply to.",
+          "items": {
+            "$ref": "#/definitions/UsersOrServicePrincipalSet"
+          }
+        },
+        "excludeServicePrincipalAsRequestor": {
+          "type": "boolean",
+          "description": "Exclude all service principals as caller from this rule."
+        },
+        "excludeServicePrincipalsAsTarget": {
+          "type": "boolean",
+          "description": "Exclude when the target of the role assignment is a service principal."
```

Comment:
Consider modelling these as enums instead of booleans 


[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)